File: discovery-provider/src/models/reaction.py
from sqlalchemy import Column, DateTime, Integer, String

from .models import Base, RepresentableMixin


class Reaction(Base, RepresentableMixin):
    __tablename__ = "reactions"

    id = Column(Integer, nullable=False, primary_key=True)
    slot = Column(Integer, nullable=False)
    reaction_value = Column(Integer, nullable=False)
    sender_wallet = Column(String, nullable=False)
    reaction_type = Column(String, nullable=False)
    reacted_to = Column(String, nullable=False)
    timestamp = Column(DateTime, nullable=False)
    tx_signature = Column(String, nullable=True)

File: packages/identity-service/build_test/src/models/reaction.js
'use strict';
module.exports = (sequelize, DataTypes) => {
    const Reactions = sequelize.define('Reactions', {
        id: {
            allowNull: false,
            autoIncrement: true,
            primaryKey: true,
            type: DataTypes.INTEGER
        },
        slot: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        reactionValue: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        senderWallet: {
            type: DataTypes.STRING,
            allowNull: false
        },
        reactedTo: {
            type: DataTypes.STRING,
            allowNull: false
        },
        reactionType: {
            type: DataTypes.STRING,
            allowNull: false
        },
        createdAt: {
            allowNull: false,
            type: DataTypes.DATE
        },
        updatedAt: {
            allowNull: false,
            type: DataTypes.DATE
        }
    });
    return Reactions;
};
